docs/releasing.md

```diff
@@ -0,0 +1,138 @@
+# Releasing ironic-image
+
+This document explains how to prepare and publish a release for ironic-image, and
+how often a new release branch is created.
+
+**NOTE**: Always follow
+[release documentation from the main branch](https://github.com/metal3-io/ironic-image/blob/main/docs/releasing.md).
+Release documentation in release branches may be outdated.
+
+## Pre-release actions
+
+Things you should check before making a release:
+
+- Check the
+  [Metal3 release process](https://github.com/metal3-io/metal3-docs/blob/main/processes/releasing.md)
+  for high-level process and possible follow-up actions
+- Verify the latest bugfix or stable branches (which is the most recent) in ironic
+  and ironic-inspector upstream repositories;
+- Verify compatibility with latest sushy and ironic-lib upstream releases or master
+  branches based on ironic and ironic-inspector bugfix or stable requirements
+  and constraints
+- Verify openstack upper-constraints compatibility with ironic and ironic-inspector
+  bugfix or stable branches
+- Verify any other direct or indirect dependency is uplifted to close any public
+  vulnerabilities
+
+  ## Permissions
+
+Creating a release requires repository `write` permissions for:
+
+- Tag pushing
+- Branch creation
+- GitHub Release publishing
+
+These permissions are implicit for the org admins and repository admins.
+A release team member gets permissions via `metal3-release-team`
+membership. This GitHub team has the required permissions in each repository
+to make a release. Adding person to the team provides the necessary
+rights  in all relevant repositories in the organization. Individual persons
+should not be given permissions directly.
+
+## Process
+
+ironic-image uses [semantic versioning](https://semver.org). For version `vX.Y.Z`:
+
+### Repository setup
+
+Clone the repository:
+`git clone git@github.com:metal3-io/ironic-image`
+
+or if using existing repository, verify your intended remote is set to
+`metal3-io`: `git remote -v`. For this document, we assume it is `origin`.
+
+- If creating a new release branch, identify the commit you wish to create the
+  branch from, and create a branch `release-X.Y` where `X.Y` corresponds to the
+  ironic upstream release version, for example for `ironic 22.1` it's `release-22.1`:
+  `git checkout <sha> -b release-X.Y` and push it to remote:
+  `git push origin release-X.Y` to create it
+- If creating a new patch release, use existing branch `release-X.Y`:
+  `git checkout origin/release-X.Y`
+- Wait for the CI support to be configured, then in the newly created release branch:
+   - Add a copy of upper-constraints.txt from [Openstack Requirements](https://opendev.org/openstack/requirements)
+     to replace the placeholder
```

Comment:
A few more words here probably? Like, if we pin ironic to a stable branch, the requirements should also come from the same stable branch. For bugfix branches... just whatever the current version is?